### PR TITLE
Temporal: Porting some more of proposal-temporal tests

### PR DESCRIPTION
**Pull request by ptomato**
Recommend reviewing commit by commit; the tests within each commit are very similar.

### CODE REVIEW

File: test/built-ins/Temporal/ZonedDateTime/prototype/with/offset-property-invalid-string.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.zoneddatetime.prototype.equals
```
Comment (Aditi-1400) [inline test/built-ins/Temporal/ZonedDateTime/prototype/with/offset-property-invalid-string.js:5]: Should be: `esid: sec-temporal.zoneddatetime.prototype.with`

**Comment by Aditi-1400** [comment]
Thanks. 
I didn't find anything else, looks good to me! 👍🏽

File: test/built-ins/Temporal/Calendar/prototype/weekOfYear/argument-propertybag-calendar-string.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+features: [Temporal]
+---*/
+
+const instance = new Temporal.Calendar("iso8601");
+
+const calendar = "iso8601";
+
+const arg = { year: 1976, monthCode: "M11", day: 18, calendar };
+const result = instance.weekOfYear(arg);
+assert.sameValue(result, 47, `Calendar created from string "${arg}"`);
```
Comment (jessealama) [inline test/built-ins/Temporal/Calendar/prototype/weekOfYear/argument-propertybag-calendar-string.js:16]: Perhaps we should some something more like 

````
`Calendar created from ${typeof arg} "${arg}"`
````

? In this case, I guess it's not a string.

Comment (ptomato) [reply test/built-ins/Temporal/Calendar/prototype/weekOfYear/argument-propertybag-calendar-string.js:16]: I think I intended it to be `Calendar created from string ${calendar}` in all these tests.

Comment (ptomato) [reply test/built-ins/Temporal/Calendar/prototype/weekOfYear/argument-propertybag-calendar-string.js]: (at least, the property bag ones)

**Comment by ptomato** [comment]
Thanks for the reviews. I've pushed an update.

**Review by Aditi-1400 — APPROVED** [review]

**Pull request merged**